specs/specTemplate/sampleComponentSpec.md

```diff
@@ -1,56 +1,95 @@
 **Table of Contents**
 
-- [Definition](#definition)
+- [Description](#description)
+
 - [Elements](#elements)
-- [Props](#props)
-- [Style](#style)
+
+- [API](#api)
+
+- [States](#states-(if-applicable))
+
 - [Accessibility](#accessibility)
+
 - [Behavior](#behavior)
+  - [Validation](validation)
   - [Edge case handling](edge-case-handling)
+
+- [Input Methods](#input-methods)
   - [Keyboard](#keyboard)
   - [Mouse](#mouse)
   - [Touch](#touch)
-- [Examples](#examples)
+
+- [RTL](#rtl)
+
+- [DOM Structure](#dom-structure)
+
 - [Design](#design)
 
   ​
 
-## Definition
+
+
+## Description
 
 Brief component description
 
-> e.g.
+> E.g.
 > **Slider** is a component that allows users to input or select value from a range.
 > Sliders are great for adjusting settings that reflect intensity levels (volume, brightness, color saturation).
 
 
 
-## Elements
+### Elements
 
-Screenshot & brief elements description
+All elements of the component and their interactivity will be described, including a visual representation of it (image or animations).
 
 > e.g.
-> Screenshot of the component (http://joxi.ru/bmonQewtMy479r)
-> **Slider consists of:** "bar", "handle" and "progress". The "handle" is dragged across the "bar" in order to give the slider a desired value while "progress" marks range from min value to current value. 
-
+>
+> ![elements](./assets/elements.png)
+> **Toggle** consists of:  **switch**, **icon**, and **background**. The **switch** moves across the **background** in order to set the value to On/Off, and **icon** indicates the state of the value.
 
 
-## Props
 
-See [README.md](./README.md) for more info. 
+## API
 
+**Props**
 
+See [README.md](./README.md) for more info. 
 
-## Style
+**Style**
 
 Brief description of pseudo-classes and custom CSS states that can be applied to the component.
 See [README.md](./README.md) for more info. 
 
 
 
+## States (if applicable)
+
+A description of all the internal states of the component, and how they should be shown visually.
+
+| State   | Description                              | Link to design |
+| :------ | :--------------------------------------- | -------------- |
+| Default | Default component appearance             |                |
+| Hover   | User hovered over bar OR handle          |                |
+| …       |                                          |                |
+| Error   | Error state for the component (can be set with :error pseudo-class) |                |
+
+Design [assets](https://zpl.io/2kRTvO)
+
+
+
 ## Accessibility
 
-Describe how do we cover accessibility issues in current component
+Our components should be accessible according to [WAI-ARIA 1.0](https://www.w3.org/TR/wai-aria/) standards.
+
+We follow these guidelines when [using ARIA](https://www.w3.org/TR/using-aria/#intro)
```

Comment:
@arnonkehat yes. This is what I meant. Let's finalize these docs as they are. Then you'll put them into your doc center. And then Gilad will be able to work on Accessibility section.